be/src/storage/inverted/clucene/clucene_inverted_writer.cpp

```diff
@@ -121,8 +122,25 @@ class CLuceneInvertedWriterImpl : public CLuceneInvertedWriter {
         if constexpr (is_string_type(field_type)) {
             auto* _val = (Slice*)values;
             for (int i = 0; i < count; ++i) {
-                new_fulltext_field(_val->data, _val->size);
-                _index_writer->addDocument(_doc.get());
+                const char* s = _val->data;
+                size_t size = _val->size;
+                if (size > _wstring_buffer.capacity()) {
+                    _wstring_buffer.reserve(size);
+                }
+                TCHAR* data = const_cast<TCHAR*>(_wstring_buffer.data());
+                std::mbstowcs(data, s, size);
```

Comment:
this will write data after its _wstring.size(), even though the underlying buffer has enough space. It is not suggested to do it in this hacked way that completely breaks the std::string semantics.

Reply:
same as above